<?xml version = "1.0"?>
<launch>
     <!-- Global namespace -->
     <arg name = "ns"
          default = "linear_feedback_controller" />
     <!-- Load the linear feedback controller. -->
     <arg name = "spawn"
          default = "true" />

     <!-- Adding advanced debugging logs. -->
     <arg name = "debug_log"
          default = "false" />

     <!-- Use a default set of parameters for the lfc -->
     <arg name = "default_params"
          default = "false" />

     <!-- Settings for initializing the Talos params. -->
     <arg name = "simulation"
          default = "false" />
     <arg name = "activate_left_arm"
          default = "false" />
     <arg name = "activate_left_wrist"
          default = "false" />
     <arg name = "activate_right_arm"
          default = "false" />
     <arg name = "activate_right_wrist"
          default = "false" />
     <arg name = "activate_torso"
          default = "false" />
     <arg name = "activate_head"
          default = "false" />
     <arg name = "activate_left_leg"
          default = "false" />
     <arg name = "activate_right_leg"
          default = "false" />

     <!-- Activate DEBUG streams -->
     <group if = "$(arg debug_log)">
       <env name="ROSCONSOLE_CONFIG_FILE"
        value="$(find linear_feedback_controller)/tests/rosconsole.conf"/>
       <env name="CTEST_OUTPUT_ON_FAILURE" value="1" />
     </group>

     <!-- Load hardware configuration. -->
     <include file = "$(find linear_feedback_controller)/launch/talos_params.launch">
          <arg name = "ns"
               value = "$(arg ns)" />
          <arg name = "simulation"
               value = "$(arg simulation)" />
          <arg name = "activate_left_arm"
               value = "$(arg activate_left_arm)" />
          <arg name = "activate_left_wrist"
               value = "$(arg activate_left_wrist)" />
          <arg name = "activate_right_arm"
               value = "$(arg activate_right_arm)" />
          <arg name = "activate_right_wrist"
               value = "$(arg activate_right_wrist)" />
          <arg name = "activate_torso"
               value = "$(arg activate_torso)" />
          <arg name = "activate_head"
               value = "$(arg activate_head)" />
          <arg name = "activate_left_leg"
               value = "$(arg activate_left_leg)" />
          <arg name = "activate_right_leg"
               value = "$(arg activate_right_leg)" />
     </include>

     <!-- Load default parameters for the lfc. -->
     <group if = "$(arg default_params)">
          <rosparam command = "load"
                    file = "$(find linear_feedback_controller)/config/talos_linear_feedback_controller_params.yaml" ns="$(arg ns)"/>
     </group>


     <!-- It has been shown that the example-robot-data urdf works kinda better on the robot than the one loaded from PAL's package. -->
     <param name="$(arg ns)/robot_description" textfile="$(find example-robot-data)/robots/talos_data/robots/talos_full_v2.urdf"/>
     <!-- For our controller we use the srdf of the example robot data as it contain the half-sitting posture. -->
     <param name="$(arg ns)/robot_description_semantic" textfile="$(find example-robot-data)/robots/talos_data/srdf/talos.srdf"/>

     <!-- Load the linear feedback controller. -->
     <group if = "$(arg spawn)">
          <node name = "linear_feedback_controller_spawner"
                pkg = "controller_manager"
                type = "spawner"
                output = "screen"
                args = "linear_feedback_controller" />
     </group>

</launch>
